crates/cow-amm/src/cow_amm_constant_product_factory.rs

```diff
@@ -0,0 +1,30 @@
+use {
+    crate::{cow_amm::CowAmm, indexer::CowAmmRegistry},
+    shared::impl_event_retrieving,
+};
+
+impl_event_retrieving! {
+    pub Contract for contracts::cow_amm_constant_product_factory
+}
+
+pub struct CowAmmConstantProductFactoryHandler;
+
+impl CowAmmConstantProductFactoryHandler {
+    /// Apply the event to the given CoW AMM registry
+    pub async fn apply_event(
+        event: &contracts::cow_amm_constant_product_factory::Event,
+        cow_amms: &mut CowAmmRegistry,
```

Comment:
`CowAmmRegistry` also likely has to be generic over the specific type that implements the `CowAmm` trait to work generically with all implementations.